aten/src/ATen/native/cuda/DilatedMaxPool3d.cu

```diff
@@ -263,45 +347,62 @@ void max_pool3d_with_indices_out_cuda_template(
     otime, oheight, owidth,
     "max_pool3d_with_indices_out_cuda_template()");
 
+  bool channels_last = input.ndimension() == 5 && input.suggest_memory_format() == at::MemoryFormat::ChannelsLast3d;
+
   if (input.ndimension() == 4) {
     output.resize_({ nslices, otime, oheight, owidth});
```

Comment:
shouldn't you also treat 4d tensors in the same way (by unsqueezing them to have batch dimension first, and treating them as channels-last if appropriate?)

Reply:
Added, although in this process I encountered some suspicious behavior from `suggest_memory_format` where the channels-last batch size 1 input after going through squeeze and unsqueeze was suggested as `contiguous` even though `is_contiguous` was `false` and `is_contiguous(at::MemoryFormat::ChannelsLast3D)` was `true` after the unsqueeze.